berlekamp_welch.go

```diff
@@ -87,10 +96,11 @@ func (fc *FEC) Correct(shares []Share) error {
 	// the syndrome matrix.
 	synd, err := fc.syndromeMatrix(shares)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	buf := make([]byte, len(shares[0].Data))
 
+	badShareMap := make(map[int]bool)
```

Comment:
This is actually a set (implemented via map), so `badSharesSet` would be a better name.